src/nvim/normal.c

```diff
@@ -3940,6 +3957,101 @@ void scroll_redraw(int up, long count)
   redraw_later(VALID);
 }
 
+///
+/// Return length of line "lnum" for horizontal scrolling.
+///
+static colnr_T scroll_line_len(linenr_T lnum)
+{
+  char_u *p = ml_get(lnum);
+  colnr_T col = 0;
+  if (*p != NUL) {
+    for (;;) {
+      int w = chartabsize(p, col);
+      mb_ptr_adv(p);
+      if (*p == NUL) {    // don't count the last character
+        break;
+      }
+      col += w;
+    }
+  }
+  return col;
+}
+
+///
+/// Find longest visible line number.  If this is not possible (or not desired,
+/// by setting 'h' in "guioptions") then the current line number is returned.
```

Comment:
You removed the check for 'guioptions' so let's remove that sentence from this comment. Also makes sense to remove the `gui_` prefix from the function name now.